experimental/delegate/mlp_aie_bf16_plugin.cpp

```diff
@@ -256,33 +421,18 @@ int setupNPUAccelerator() {
 int aie_matmul(Params *params) {
     std::cout << "[AIE Delegate]: Computing AIE matmul of " << params->getShapeStr() << std::endl;
     int cnt = 0;
-
-    // quantize and copy weights to XRT BO
     auto xrtState = XrtState::getInstance();
-    A_DATATYPE *bufA = xrtState->boA.map<A_DATATYPE *>();
-    // std::cout << "Input A" << std::endl;
-    for (int i = 0; i < M; i++) {
-        // std::cout << '[';
-        for (int j = 0; j < K; j++) {
-            float f = params->lhs.getElement(i, j, K);
-            bfloat16_t bf = toBfloat16(f);
-            // std::cout << "{f:" << f << ",bf:" << bf << "}";
-            *(bufA + i * K + j) = bf;
-        }
-        // std::cout << "]," << std::endl;
-    }
 
-    cnt = 0;
-
-    // quantize and copy input to XRT BO
+    // Copy inputs to kernel input BOs
+    A_DATATYPE *bufA = xrtState->boA.map<A_DATATYPE *>();
+    TensorCopier<ModelLhsDType, A_DATATYPE>::copy(bufA, params->lhs.get(), aVolume);
```

Comment:
I believe none of these copies should be necessary with this API call, youi can just mapo your host buffer to xrt buffer?
https://xilinx.github.io/XRT/master/html/xrt_native.main.html#_CPPv4N3xrt2bo2boERKN3xrt6deviceEPv6size_tN2bo5flagsE12memory_group